rllib/policy/sample_batch.py

```diff
@@ -252,7 +256,7 @@ def __init__(self, *args, **kwargs):
             # TODO: Drop support for lists and Numbers as values.
             # Convert lists of int|float into numpy arrays make sure all data
             # has same length.
-            if isinstance(v, (Number, list)):
+            if isinstance(v, (Number, list)) and not k == SampleBatch.INFOS:
```

Comment:
Since infos are always dicts, listyfying them messes with later steps of adding batch dimensions inside of them etc.